tests/unit_tests/ensemble_evaluator/test_ensemble_evaluator.py

```diff
@@ -177,22 +183,30 @@ async def test_new_monitor_can_pick_up_where_we_left_off(evaluator):
             {identifiers.ERROR_MSG: "error"},
         )
 
-    # we have to wait for the batching dispatcher to process the events, and for the
-    # internal ensemble state to get updated before connecting and getting a full
-    # ensemble snapshot
-    while len(evaluator._dispatcher._buffer) > 0:
-        await asyncio.sleep(0.1)
+    def check_if_final_snapshot_is_complete(final_snapshot: Snapshot) -> bool:
+        try:
+            assert final_snapshot.status == ENSEMBLE_STATE_UNKNOWN
+            assert (
+                final_snapshot.get_job("0", "0").status == FORWARD_MODEL_STATE_RUNNING
+            )
+            assert (
+                final_snapshot.get_job("1", "0").status == FORWARD_MODEL_STATE_FINISHED
+            )
+            assert (
+                final_snapshot.get_job("1", "1").status == FORWARD_MODEL_STATE_FAILURE
+            )
+            return True
+        except AssertionError:
+            return False
+
     # reconnect new monitor
     async with Monitor(config_info) as new_monitor:
-        new_events = new_monitor.track()
-        full_snapshot_event = await new_events.__anext__()
-
-        assert full_snapshot_event["type"] == identifiers.EVTYPE_EE_SNAPSHOT
-        snapshot = Snapshot(full_snapshot_event.data)
-        assert snapshot.status == ENSEMBLE_STATE_UNKNOWN
-        assert snapshot.get_job("0", "0").status == FORWARD_MODEL_STATE_RUNNING
-        assert snapshot.get_job("1", "0").status == FORWARD_MODEL_STATE_FINISHED
-        assert snapshot.get_job("1", "1").status == FORWARD_MODEL_STATE_FAILURE
+        final_snapshot = Snapshot({})
+        async for event in new_monitor.track():
+            new_snapshot = Snapshot(event.data)
+            final_snapshot.merge(new_snapshot.data())
+            if check_if_final_snapshot_is_complete(final_snapshot):
```

Comment:
No, if all asserts fail, we will wait for the next event from monitor; merge it into the snapshot, then check if the snapshot is complete. In case an event is somehow lost, the test will hang and hopefully time out.

Reply:
Then we should I guess do some timeout on the test function? `@pytest.mark.timeout(20)`

Reply:
On my Mac it takes ca. 7 seconds so 20 should be fine.

Reply:
There should be a default timeout in the pytest config in pyproject.yaml. Do you think we should have an explicit timeout instead?

Reply:
Yes, but the default is `timeout = 360` which is way to much.